src/Microsoft.Graph.Core/Requests/BaseRequest.cs

```diff
@@ -155,6 +180,56 @@ public class BaseRequest : IBaseRequest
         }
 
         /// <summary>
+        /// Sends the multipart request.
+        /// </summary>
+        /// <typeparam name="T">The expected response object type for deserialization.</typeparam>
+        /// <param name="multipartContent">The multipart object to send.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> for the request.</param>
+        /// <param name="completionOption">The <see cref="HttpCompletionOption"/> to pass to the <see cref="IHttpProvider"/> on send.</param>
+        /// <returns>The <see cref="HttpResponseMessage"/> object.</returns>
+        public async Task<HttpResponseMessage> SendMultiPartRequestAsync(
+            MultipartContent multipartContent,
+            CancellationToken cancellationToken,
+            HttpCompletionOption completionOption = HttpCompletionOption.ResponseContentRead)
+        {
+            if (string.IsNullOrEmpty(this.RequestUrl))
+            {
+                throw new ServiceException(
```

Comment:
Yep. That is how it should have been done from the beginning. I'd change it now except it could be a breaking for customers. We really should have a ClientException class with subtypes for validation and other error categories.